src/app/app.component.ts

```diff
@@ -283,4 +287,10 @@ export class AppComponent {
       }
     }
   }
+
+  private async checkForAppUpdatesAndApply() {
+    if (Capacitor.isNativePlatform()) {
+      await this.appUpdateService.checkForUpdatesAndAttemptImmediateUpdate();
```

Comment:
nit(non-blocking)
Would be nice to have some sort of AppConfig that can define whether to opt in to this feature, and strategy to use. E.g. `AppUpdatePrompt: {enabled: true, forceUserUpdate: false}

I'm also not sure how the UI of the feature works, but if we simply get a notification that an update is available it may be worth also providing an optional template to display when update ready